### DISCUSSION TITLE
Downloaded Video and Audio streams are not combined into one file

### DISCUSSION THREAD

**Discussion by forapreder** [setvisible/DownZemAll#203] [Q&A]
Downloaded Video and Audio streams are not combined into one file .mp4. DownZemAll did it. ArrowDL 4.0.0 mmpg lost.
![1001](https://github.com/setvisible/ArrowDL/assets/60479869/30be8ec0-ccc8-4b67-9df8-b13a48e35b23)
![1002](https://github.com/setvisible/ArrowDL/assets/60479869/37a127f5-1088-45fb-b259-851e265a8989)

**Comment by setvisible** [comment]
Hi

The merge of video and audio is managed by [youtube-dlp](https://github.com/yt-dlp/yt-dlp). 

Could you download the given example in command line with yt-dlp.exe (the one in the install folder), and check if they are merged?


Regards,

**Comment by forapreder** [comment]
I checked via the command line. I uploaded the file that is included in the program. Video and Audio are combined into one file.
You can check it yourself. This is the content.
https://revoir.tv5monde.com/toutes-les-videos/art-de-vivre/maisons-hotels-de-legende-beau-rivage-palace

**Comment by setvisible** [comment]
Hello,

I tried the example, the audio and video are downloaded separately but they are merged at the end.

I think that TV5Monde maybe doesn't give all the metadata. If so, ArrowDL gives the feeling that the file is actually downloaded while some steps aren't finished. The progress bar is based on assumed size of the file, but if TV5Monde says the file is 0 kB (ex: the audio), then the progress achieves 100% while one of the file (audio in this case) is still downloading.

=> For streaming servers that don't give all metadata, we should not trust the progress status. Instead, just wait enough time until the progress bar is dark green and/or the ETA -estimated time- is at "download finished").

If still cumbersome, maybe open a ticket on the **yt-dlp** github project, maybe they could take the TV5Monde issue into account?

Regards

**Comment by forapreder** [comment]
But DownZemAll 3.1.0 using the same yt-dlp combines Video and Audio files into one file very well. I use **portable** ArrowDL 4.0.0. Could there be a problem with **portable** version? In the portable ArrowDL 4.0.0 the matter is something else. I tried to download the file again and waited until it turned dark green, but it still downloaded two files and they were not combined. I will use portable DownZemAll 3.1.0 , It suits me quite well.
